sparse/ndsparse/_tensor.py

```diff
@@ -0,0 +1,232 @@
+        """
+        Convert tensor into scipy sparse array
+        """
+        return pt.to_sp_csr(self._tensor)
+
+    def insert(self, coord, val):
+        """
+        Inserts value into the tensor
+        """
+        self._tensor.insert(coord, val)
+
+    def order(self):
+        """
+        Returns the order of the tensor
+        """
+        return self._tensor.order()
+
+    #     def name(self):
+    #         """
+    #         Returns the name of the tensor
+    #         """
+
+    def transpose(self, new_ordering, new_format=None, name=None):
+        """
+        Returns the transpose of the tensor
+        """
+        return self._tensor.transpose(new_ordering, new_format, name)
+
+    def T():
+        """
+        Transpose with shape = [::-1]
+        """
+        return self._tensor.T()
+
+    #     def asformat():
+    #         """
+    #         """
+
+    #     def nnz():
+    #         """
+    #         """
+
+    #     def nonzero():
+    #         """
+    #         """
+    def __array__(self):
+        return np.asarray(self.todense())
+
+    def __getitem__(self, index):
+        return self._tensor[index]
+
+    def __setitem__(self, index, val):
+        self._tensor[index] = val
+
+    def __add__(self, other):
+        return pt.tensor_add(self, other, mode)
+
+    def __radd__(self, other):
+        return pt.tensor_add(other, self, mode)
+
+    def __sub__(self, other):
+        return pt.tensor_sub(self, other, mode)
+
+    def __rsub__(self, other):
+        return pt.tensor_sub(other, self, mode)
+
+    def __mul__(self, other):
+        return pt.tensor_mul(self, other, mode)
+
+    def __rmul__(self, other):
+        return pt.tensor_mul(other, self, mode)
+
+    def __truediv__(self, other):
+        return pt.tensor_div(self, other, mode)
+
+    def __rtruediv__(self, other):
+        return pt.tensor_div(other, self, mode)
+
+    def __floordiv__(self, other):
+        return pt.tensor_floordiv(self, other, mode)
+
+    def __rfloordiv__(self, other):
+        return pt.tensor_floordiv(other, self, mode)
+
+    def __ge__(self, other):
+        return pt.tensor_ge(self, other, mode)
+
+    def __gt__(self, other):
+        return pt.tensor_gt(self, other, mode)
+
+    def __le__(self, other):
+        return pt.tensor_le(self, other, mode)
+
+    def __lt__(self, other):
+        return pt.tensor_lt(self, other, mode)
+
+    def __ne__(self, other):
+        return pt.tensor_ne(self, other, mode)
+
+    def __eq__(self, other):
+        return pt.tensor_eq(self, other, mode)
+
+    def __pow__(self, power, modulo=None):
+        return pt.tensor_pow(self, power, mode)
+
+    def __abs__(self):
+        return pt.tensor_abs(self, self.format)
+
+    def __neg__(self):
+        return pt.tensor_neg(self, self.format)
+
+    def __eq__(self, other):
+        return pt.tensor_eq(self, other, mode)
+
+    def __neg__(self):
+        return pt.tensor_neg(self, self.format)
```

Comment:
This is taken care of by `NDArrayOperatorsMixin` and you can override `__array_ufunc__` instead.